### PR TITLE
data path rearrangement

### PR DESCRIPTION
**Pull request by ninarobbins**
I changed the path reading code to the new data structure suggestion: "/pathtodata/platform/level/flight_id/". It is not super clean but it works. At two points in the code (in "paths.py" and in "processor.py", a parameter `path_structure` needs to be specified (and is by default "levels_first" which adopts the new folder structure) which is not ideal. Probably there is a better way to do it but this works for now.

### CODE REVIEW

**Comment by ninarobbins** [comment]
There is an issue with the testing. I think the test should be modified accordingly (?)

**Comment by Geet-George** [comment]
Sorry, I thought the last commit would fix things, but it didn't. I am unable to reproduce the test failures locally. Don't know why :/ I'm checking.

**Review by Geet-George — COMMENTED** [review]
Any reason to change `l2_dir` to `l2dir`?

**Comment by Geet-George** [comment]
@ninarobbins for some reason, this doesn't seem to work for me, and I can't figure it out. I went through the lines, the default is "levels_first" and should give the levels first in the path. But when the test actually runs, this is what is logged in info:

```
2024-08-10 16:01:36,587    INFO   halodrops.helper.paths.Paths       paths.py       Line:092:
Created Path Instance: self.flight_idpath='../sample/HALO/20200101'; self.flight_id='20200101'; self.l1dir='../sample/HALO/20200101/Level_1'
2024-08-10 16:01:36,651    INFO   halodrops.helper.paths.Paths       paths.py       Line:116:
Path exists: quicklooks_path_str='../sample/HALO/20200101/Quicklooks'
```

Any idea what's happening?

**Comment by Geet-George** [comment]
:)

File: src/halodrops/processor.py [hunk 1]
```diff
@@ -901,50 +926,58 @@ def get_l2_filename(self, l2_filename: str = None):
 
         return self
 
-    def write_l2(self, l2_dir: str = None):
+    def write_l2(self, l2dir: str = None):
```
[the remark this answers is not in the document]
Comment (ninarobbins) [reply src/halodrops/processor.py:930]: was trying to stay In the same nomenclature as l0dir and l1dir attributes in the path. but maybe inappropriate!

[the remark this answers is not in the document]
Comment (ninarobbins) [reply src/halodrops/processor.py:930]: sorry!

**Comment by ninarobbins** [comment]
> :) Passes the test! But why shouldn't there be `flight_id`s?

fixed! I missed to write the correct date and it was failing, now I think
 it's okay

**Comment by ninarobbins** [comment]
> :) Passes the test! But why shouldn't there be `flight_id`s?
 not sure why I did that... I put it back now

**Comment by ninarobbins** [comment]
@Geet-George would it be a good idea to add the type of structure to the config file?

**Comment by Geet-George** [comment]
> @Geet-George would it be a good idea to add the type of structure to the config file?

You mean mention in the config file if it is levels first or flights first? If so, that can be provided by the `path_structure` argument, right? I think, we should:

- [ ] clarify in docs `levels_first` is recommended and default.

**Pull request closed without merging**